From e0b461db2cc5a9c2e22099af450fd41edeeec5be Mon Sep 17 00:00:00 2001
From: Davor Joja <davorjoja@logicbricks.com>
Date: Tue, 18 Nov 2014 20:03:55 +0100
Subject: [PATCH 465/793] gpu: drm: xylon: Registered plane update

Set crtc base by calling registered plane driver function.
This removes direct plane driver function calls and removes functionality
redundancy.

Signed-off-by: Davor Joja <davorjoja@logicbricks.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Tested-by: Radhey Shyam Pandey <radheys@xilinx.com>
Tested-by: Christian Kohn <christian.kohn@xilinx.com>
---
 drivers/gpu/drm/xylon/xylon_crtc.c |   21 ++++++++++-----------
 1 files changed, 10 insertions(+), 11 deletions(-)

diff --git a/drivers/gpu/drm/xylon/xylon_crtc.c b/drivers/gpu/drm/xylon/xylon_crtc.c
index 4012929..012882e 100644
--- a/drivers/gpu/drm/xylon/xylon_crtc.c
+++ b/drivers/gpu/drm/xylon/xylon_crtc.c
@@ -203,24 +203,23 @@ static int xylon_drm_crtc_mode_set_base(struct drm_crtc *base_crtc,
 					int x, int y,
 					struct drm_framebuffer *old_fb)
 {
+	const struct drm_plane_funcs *funcs = base_crtc->primary->funcs;
 	int ret;
 
-	ret = xylon_drm_plane_fb_set(base_crtc->primary, base_crtc->primary->fb,
-				     0, 0,
-				     base_crtc->hwmode.hdisplay,
-				     base_crtc->hwmode.vdisplay,
-				     x, y,
-				     base_crtc->hwmode.hdisplay,
-				     base_crtc->hwmode.vdisplay);
+	ret = funcs->update_plane(base_crtc->primary,
+				  base_crtc,
+				  base_crtc->primary->fb,
+				  0, 0,
+				  base_crtc->hwmode.hdisplay,
+				  base_crtc->hwmode.vdisplay,
+				  x, y,
+				  base_crtc->hwmode.hdisplay,
+				  base_crtc->hwmode.vdisplay);
 	if (ret) {
 		DRM_ERROR("failed set plane mode\n");
 		return ret;
 	}
 
-	xylon_drm_plane_commit(base_crtc->primary);
-
-	xylon_drm_crtc_dpms(base_crtc, DRM_MODE_DPMS_ON);
-
 	return 0;
 }
 
-- 
1.7.5.4

